File: auth/src/main/java/com/harrish/auth/exception/error/AuthErrorCode.java
package com.harrish.auth.exception.error;

public enum AuthErrorCode {

    AUTH_BAD_CREDENTIALS("auth.error.bad_credentials"),
    AUTH_INVALID_TOKEN("auth.error.invalid_token"),
    AUTH_TOKEN_EXPIRED("auth.error.token_expired"),
    AUTH_ACCESS_DENIED("auth.error.access_denied");

    private final String messageKey;

    AuthErrorCode(String messageKey) {
        this.messageKey = messageKey;
    }

    public String getMessageKey() {
        return messageKey;
    }
}

File: auth-kotlin/src/main/kotlin/com/harrish/auth/exception/error/AuthErrorCode.kt
package com.harrish.auth.exception.error

enum class AuthErrorCode(val messageKey: String) {
    AUTH_BAD_CREDENTIALS("auth.error.bad_credentials"),
    AUTH_INVALID_TOKEN("auth.error.invalid_token"),
    AUTH_TOKEN_EXPIRED("auth.error.token_expired"),
    AUTH_ACCESS_DENIED("auth.error.access_denied")
}
